src/Razor/test/Microsoft.AspNetCore.Razor.VSCode.FunctionalTest/tests/GoToDefinition.test.ts

```diff
@@ -0,0 +1,101 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+ import * as assert from 'assert';
+ import { afterEach, before, beforeEach } from 'mocha';
+ import * as path from 'path';
+ import * as vscode from 'vscode';
+ import {
+     mvcWithComponentsRoot,
+     pollUntil,
+     waitForProjectReady,
+} from './TestUtil';
+
+ let cshtmlDoc: vscode.TextDocument;
+ let editor: vscode.TextEditor;
+ let cshtmlPath: string;
+
+ suite('Definition', () => {
+    before(async () => {
+        await waitForProjectReady(mvcWithComponentsRoot);
+    });
+
+    beforeEach(async () => {
+        cshtmlPath = path.join(mvcWithComponentsRoot, 'Views', 'Home', 'Index.cshtml');
+        cshtmlDoc = await vscode.workspace.openTextDocument(cshtmlPath);
+        editor = await vscode.window.showTextDocument(cshtmlDoc);
+    });
+
+    afterEach(async () => {
+        await vscode.commands.executeCommand('workbench.action.revertAndCloseActiveEditor');
+        await pollUntil(() => vscode.window.visibleTextEditors.length === 0, 1000);
+    });
+
+    test('Definition of injection gives nothing', async () => {
+        const imports = path.join(mvcWithComponentsRoot, 'Views', '_ViewImports.cshtml');
```

Comment:
Any reason to not just use the `editor` defined above? I ask because technically in _ViewImports you aren't allowed to have implicit expressions (`@DateTime...`) or HTML markup. Ditto with the inside file test.